### PR TITLE
Checks for SLES12 to enable Storage repositories

### PR DESCRIPTION
**Pull request by dguitarbite**

### CODE REVIEW

**Comment by toabctl** [comment]
More comments here? Would be good to get this merge to continue with the Storage product integration.

File: chef/cookbooks/provisioner/libraries/repositories.rb [hunk 1]
```diff
@@ -101,6 +127,8 @@ def get_repos(provisioner_server_node, platform, version)
             )
           when "12.0"
             repo_names = %w(
+              SUSE-Storage-1.0-Pool
+              SUSE-Storage-1.0-Updates
```
Comment (vuntz) [inline chef/cookbooks/provisioner/libraries/repositories.rb]: I don't think these two lines are correct. They mean that the repos are always required, which defeats the rest of this pull request.

Comment (dguitarbite) [inline chef/cookbooks/provisioner/libraries/repositories.rb]: As per my previous discussions, I thought this was the requirement. I will talk to Dirk once.

Comment (dirkmueller) [inline chef/cookbooks/provisioner/libraries/repositories.rb]: No, the Storage dependency is optional, its only required for Ceph deployments.

**Comment by dguitarbite** [comment]
@toabctl This code should work. If you want to test this, I suggest you to use the flag `NOQACROWBARDOWNLOAD=1` and also to edit this line https://github.com/SUSE-Cloud/automation/blob/master/scripts/qa_crowbarsetup.sh#L1227 which unsets variable `wantceph` bydefault.

File: chef/cookbooks/provisioner/libraries/repositories.rb [hunk 2]
```diff
@@ -52,24 +62,40 @@ def inspect_repos(node)
...
-            node.set[:provisioner][:suse][:missing_hae] = missing
-            node.save
+          if node[:provisioner][:suse][:missing_hae] != missing_hae
+            node.set[:provisioner][:suse][:missing_hae] = missing_hae
+          end
+          if node[:provisioner][:suse][:missing_storage] != missing_storage
+            node.set[:provisioner][:suse][:missing_storage] = missing_storage
           end
+
+          node.save
```
Comment (vuntz) [inline chef/cookbooks/provisioner/libraries/repositories.rb]: Can you save only if there was a `node.set`? (Just use a variable to know if there's a need for saving)

Comment (dguitarbite) [inline chef/cookbooks/provisioner/libraries/repositories.rb]: Yes, I will update the code.

File: chef/cookbooks/provisioner/libraries/repositories.rb [hunk 3]
```diff
@@ -52,22 +62,41 @@ def inspect_repos(node)
...
           # set an attribute about missing repos so that cookbooks and crowbar
           # know that HA cannot be used
+          # know that SUSE_Storage cannot be used
+          node_set=false
           node.set[:provisioner][:suse] ||= {}
-          if node[:provisioner][:suse][:missing_hae] != missing
-            node.set[:provisioner][:suse][:missing_hae] = missing
+          if node[:provisioner][:suse][:missing_hae] != missing_hae
+            node.set[:provisioner][:suse][:missing_hae] = missing_hae
+            node_set=true
```
Comment (vuntz) [inline chef/cookbooks/provisioner/libraries/repositories.rb]: Style nit-picking, but this should be `node_set = true` (spaces around `=`)

Comment (dguitarbite) [inline chef/cookbooks/provisioner/libraries/repositories.rb]: Thanks for catching that, my bad.

**Comment by vuntz** [comment]
+1

**Comment by dirkmueller** [comment]
+1

**Pull request merged**